Scripts/prototools.py

```diff
@@ -0,0 +1,571 @@
+
+    # Initialize the score matrix and traceback matrix
+    score_matrix = [[0] * (len_seq2 + 1) for _ in range(len_seq_on + 1)]
+    traceback_matrix = [[None] * (len_seq2 + 1) for _ in range(len_seq_on + 1)]
+
+    alignment_score = 0  # To keep track of the maximum score in the matrix
+    max_i, max_j = 0, 0  # To store the position of the maximum score
+
+    # Fill in the score matrix
+    for i in range(1, len_seq_on + 1):
+        for j in range(1, len_seq2 + 1):
+            if seq_on[i - 1] == seq2[j - 1]:
+                match_score = score_matrix[i - 1][j - 1] + match
+            else:
+                match_score = score_matrix[i - 1][j - 1] + mismatch
+
+            delete_score = score_matrix[i - 1][j] + gap
+            insert_score = score_matrix[i][j - 1] + gap
+
+            # Calculate the maximum score for the current cell
+            score = max(0, match_score, delete_score, insert_score)
+
+            # Update the score matrix and traceback matrix
+            score_matrix[i][j] = score
+
+            if score > alignment_score:
+                alignment_score = score
+                max_i, max_j = i, j
+
+            if score == match_score:
+                traceback_matrix[i][j] = "match"
+            elif score == delete_score:
+                traceback_matrix[i][j] = "delete"
+            elif score == insert_score:
+                traceback_matrix[i][j] = "insert"
+            else:
+                traceback_matrix[i][j] = "none"
+
+    # Traceback to find the aligned sequences
+    aligned_seq_on = []
+    aligned_seq2 = []
+
+    counter_identity: int = 0
+    counter_gaps: int = 0
+
+    i, j = max_i, max_j
+
+    while i >= 0 and j >= 0:
+        if i == 0 or j == 0 or traceback_matrix[i][j] == "none":
+            break
+        if traceback_matrix[i][j] == "match":
+            aligned_seq_on.append(seq_on[i - 1])
+            aligned_seq2.append(seq2[j - 1])
+            counter_identity += 1
+            i -= 1
+            j -= 1
+        elif traceback_matrix[i][j] == "delete":
+            aligned_seq_on.append(seq_on[i - 1])
+            aligned_seq2.append("-")
+            counter_gaps += 1
+            i -= 1
+        elif traceback_matrix[i][j] == "insert":
+            aligned_seq_on.append("-")
+            aligned_seq2.append(seq2[j - 1])
+            counter_gaps += 1
+            j -= 1
+
+    # Reverse the aligned sequences
+    aligned_seq_on = "".join(aligned_seq_on[::-1])
+    aligned_seq2 = "".join(aligned_seq2[::-1])
+
+    alignment_length = (len(aligned_seq_on)
+                        if len(aligned_seq_on) < len(aligned_seq2)
+                        else len(aligned_seq2))
+
+    try:
+        identity = round(counter_identity/alignment_length, 4)
+    except ZeroDivisionError:
+        print('It appears the alignment length of your sequences is 0.')
+        identity = 'NaN'
+        prettify = False
+
+    # Prettify an alignment output
+    if prettify is True:
```

Comment:
```suggestion
    if prettify:
```
Я про это кстати тоже комментировал:)